Fix login and registration events

DFurnes opened the pull request:
#### Changes

Fixes #5119. Since the `dosomething_user_form_alter` and `dosomething_user_preprocess_page` hooks are called on every page load (as we prepare that form on every page load), this event was constantly being triggered. This PR moves that logic in to the appropriate hooks for the user login and registration events.
#### How should this be tested?

I tested on my local box by adding a `dpm` call to [`dosomething_helpers_add_analytics_event`](https://github.com/DFurnes/phoenix/blob/d0627f7e9a2aebf64f877138987c59bab4998ee9/lib/modules/dosomething/dosomething_helpers/dosomething_helpers.module#L521-531). Everything seemed to function correctly when logging in or registering from either the header or the campaign pitch page.
#### Questions

Because `hook_user_login` is also called after registering a new user, registration will trigger both an `Authentication - Register` and an `Authentication - Login` event on the subsequent page load. I think this makes sense and keeps the implementation from getting too hairy... does anyone see any problems?

---

For review: @angaither

DFurnes committed:
Remove unused variable.

These variables are defined, but never used in this scope
(dosomething_user_form_alter).

DFurnes committed:
Add login & register custom events using hook system.

Use Drupal’s `hook_user_login` and `hook_user_insert` to add custom
events to the following page load on login/registration.

angaither commented on lib/modules/dosomething/dosomething_user/dosomething_user.module:
```diff
@@ -698,6 +699,11 @@ function dosomething_user_user_insert(&$edit, &$account, $category = NULL) {
+  // Trigger an analytics event on registration
+  // @NOTE: The 'Authentication - Login' event will *also* be triggered here,
+  // since registered users are subsequently logged in to their new accounts.
+  dosomething_helpers_add_analytics_event('Authentication', 'Register');
```
instead of doing this in `hook_insert()` maybe we can do it in the `new_user` submit handler it was in before. it's not a huge deal, but since it's not changing the user object, and it doesn't need to be in the insert hook. just a thought.

angaither replied:
that one _should_ only fire from the `user/register` event

DFurnes replied:
Hmm, I looked at the `new_user` method but it seemed to be mostly concerned with transactional logic. What's the benefit of doing things in that method over the `user_insert` hook?

DFurnes replied:
Ah, it looks like `user_insert` is intended more for database changes? If so, that makes sense to move it. :+1:

DFurnes replied:
Updated in bc092e8. :hammer:

DFurnes committed:
Move analytics hook into form submit handler.

This makes more sense, since hook_insert_user seems to be
intended more for database-related changes. I've also split
message broker logic into a new "private" method for better
readability.

angaither commented:
:+1: lovely, nice new function too!
